### DISCUSSION TITLE
accessing data on the screen from a modal

### DISCUSSION THREAD

**Discussion by ablbol** [rrousselGit/riverpod#1288] [Q&A]
Hi, 
I have a profile variable in a screen which launches  modal1 through a button, and modal1 will launch modal2 through a button.  Both modals need access to the profile variable defined in the screen. What is the best way to do that using Riverpod?  I don't want to you StateProvider.  I prefer to user Provider. How can dependencies help me in this situation?

Thanks

**Comment by rrousselGit** [comment]
Assuming you're using Riverpod 2.0.0-dev, you can use ProviderScope's `parent` parameter

See the changelog & doc of ProviderScope for an example
